### ISSUE TITLE
BPM limitation

### ISSUE THREAD

**Issue by jean-barre** [adamstark/BTrack#9]
I've been struggling to retrieve the BPM of Eagles - Hotel California: 75.
Until I found out your algorithm makes sure the tempo is comprised between 80 and 160.
Could you please explain this restriction?

**Comment by adamstark** [comment]
Good question! In short, there is an ambiguity in which "metrical level" to use when tapping in time to a piece of music. Hotel California could be at 75 bpm or it could be twice that speed at 150 bpm. It is a different interpretation of the piece of music. And to an algorithm with no musical sensibilities, 300bpm, 600bpm and 37.5 bpm are all equally valid choices.

So, when designing the algorithm, we had to deal with this question - is it 75 bpm or 150 bpm? We chose to limit all tempo estimations to be between a "tempo octave" of 80 bpm and 160bpm. If an estimate came down to 75bpm, we would always pick a value a tempo octave above (twice as fast). This way, although it may not satisfy someone wanting to use 75bpm, at least the algorithm is consistent in its output. Some other beat tracking algorithms do not do this, and have a problem where they would switch between faster and slower estimates with no clear reason as to why.

Further to this, theoretically, one could build an application that extracts a 75bpm tempo/beat framework from the 150bpm estimate of the algorithm.

So yes, I hope that makes sense. It was a choice to remove ambiguity of choice in tempo estimation.

Adam
